DEVELOPMENT.md

```diff
@@ -0,0 +1,101 @@
+# Developing Obot
+
+What follows is a rundown on different ways to run and develop Obot, its UI and its tools locally.
+
+## Running Obot
+
+The easiest way to run Obot locally is to run `make dev`. This will launch three processes: the API server, admin UI, and user UI. Opening `http://localhost:8080/admin/` will launch the admin UI. Changing the UI code will update the UI automatically. Changing any of the Go code requires restarting the server.
+
+## Building and Running the Obot Docker Image
+
+Obot is ultimately packaged into an image for distribution. You can build said image with `docker build -t my-obot .`, and then run the image via `docker run -p 8080:8080 my-obot`.
+
+## Debugging Obot
+
+It is possible to run the server and/or UIs in and IDE for debugging purposes. These steps layout what is necessary for Jet Brains IDEs, but an equivalent process can be used with VSCode based editors.
+
+### Server
+
+To run the server in GoLand:
+1. Create a new "Go Build" configuration.
+2. In the "Program Arguments" section, enter `server --dev-mode`.
+
+Then you're ready to run or debug this target.
+
+### Admin UI
+
+To run the Admin UI in GoLand or Web Storm:
+1. Create a new "npm" build.
+2. In the "package.json" dropdown, select the `package.json` file in the `ui/admin` directory.
+3. In the "Command" dropdown, select `run`.
+4. In the "Scripts" dropdown, select `dev`.
+5. In the "Environment" section, enter `VITE_API_IN_BROWSER=true`.
+
+Then you're ready to run or debug this target.
+
+### User UI
+
+To run the User UI in GoLand or Web Storm:
+1. Create a new "npm" build.
+2. In the "package.json" dropdown, select the `package.json` file in the `ui/user` directory.
+3. In the "Command" dropdown, select `run`.
+4. In the "Scripts" dropdown, select `dev`.
+5. In the "Environment" section, enter `VITE_API_IN_BROWSER=true`.
+
+Then you're ready to run or debug this target.
+
+## Developing Obot Tools
+
+Obot has a set of packaged tools. These tools are in the repo `github.com/obot-platform/tools`. By default, Obot will pull the tools from this repo. However, when developing tools in this repo, you can follow these steps to use a local copy.
+
+1. Clone `github.com/obot-platform/tools` to your local machine.
+2. In the root directory of the tools repo on your local machine, run `make build`.
+3. Run the Obot server, either with `make dev` or in your IDE, with the `OBOT_SERVER_TOOL_REGISTRY` environment variable set to the root directory of the tools repo.
+
+Now, any time one of these tools is run, your local copy will be used.
+
+> [!IMPORTANT]
+> Any time you change a Go based tool in your local repo, you must run `make build` in the tools repo for the changes to take effect with Obot.
```

Comment:
Well I usually only build that binary specifically for the tool I care instead of building all of it.